src/molecules/formfields/GoogleAutoCompleteField/__tests__/index.spec.js

```diff
@@ -106,6 +106,53 @@ describe('<GoogleAutoCompleteField />', () => {
       expect(props.autocompleteAddressFields.mock.calls.length).toBe(1);
       expect(props.autocompleteAddressFields.mock.calls[0][0]).toEqual(expectedData);
     });
+
+    describe('when google does not provide street_number', () => {
+      let addressValuesWithoutStreetNumber;
+      let state;
+
+      beforeEach(() => {
+        addressValuesWithoutStreetNumber = addressValues
+          .filter(value => !value.types.includes('street_number'));
+
+        state = {
+          autocomplete: {
+            getPlace: jest.fn(() => ({
+              address_components: addressValuesWithoutStreetNumber,
+            }))
+          }
+        };
+      });
+
+      it('should extract the street_number manually', () => {
+        const streetNumber = 1234;
+        const id = 'autocomplete-id';
+        const input = { name: 'address', value: `${streetNumber} Broadw` };
+        const wrapper = shallow(<GoogleAutoCompleteField {...props} {...{ id, input }} />);
+
+        wrapper.setState(state);
+        wrapper.instance().updateAddress();
+
+        expect(state.autocomplete.getPlace.mock.calls.length).toBe(1);
+        expect(props.autocompleteAddressFields.mock.calls.length).toBe(1);
+        expect(props.autocompleteAddressFields.mock.calls[0][0])
+          .toEqual({ ...expectedData, number: streetNumber });
+      });
+
+      it('should not return a street_number, if extraction fails', () => {
+        const id = 'autocomplete-id';
+        const input = { name: 'address', value: 'Twelve Broadway' };
+        const wrapper = shallow(<GoogleAutoCompleteField {...props} {...{ id, input }} />);
+
+        wrapper.setState(state);
+        wrapper.instance().updateAddress();
+
+        expect(state.autocomplete.getPlace.mock.calls.length).toBe(1);
+        expect(props.autocompleteAddressFields.mock.calls.length).toBe(1);
+        expect(props.autocompleteAddressFields.mock.calls[0][0])
+          .toEqual({ ...expectedData, number: undefined });
```

Comment:
Actually should we explicitly return `null` instead?

Reply:
I'm not sure... I left it as `undefined` because that's how the original implementation worked if a street number were missing, and I wasn't sure if downstream code made any assumptions about non-present values. It's _probably_ safe though...

Reply:
Gotcha! I personally like explicitly returning `null` instead of depending on the implicit return value. I think it sends a stronger signal that we are expecting this falsey value. An `undefined` can be a bit vague on whether the falsey value is intentionally being sent down or is coming unexpectedly.

Reply:
Lemme see if it causes anything to misbehave

Reply:
Well actually, I think omitting the `number` value _is_ the correct approach, because in this component you can pass in the fields you want to receive off of the google results, and if the street number isn't present there then it should be missing (aka `undefined`) rather than present but equal to `null`. In this test I was using `{ ...expectedData, number: undefined }` as a shorthand for
```js
omit(expectedData, 'number')
```
I'll update the test to make it more clear.

Reply:
oohh i see - thanks for the clarification!